flywheel_common/storage/__init__.py

```
@@ -0,0 +1,37 @@
+"""Returns appropriate storage provider based on the url provided"""
+from __future__ import print_function
+
+from pkg_resources import iter_entry_points
+
+from .interface import Interface
+from .util import path_from_uuid, format_hash, parse_storage_url
+
+STORAGE_TYPES = {}
+LOADED_TYPES = {}
+DEFAULT_TYPE = None
+
+def create_flywheel_fs(url, default='osfs'):
+    """
+    This loads the storage provider based on the url provided
+    """
+    if not STORAGE_TYPES:
```

If there end up being no storage entry points, then we would loop every time we enter this function. Consider using `None` to represent the uninitialized state.

---

I did consider this and if we have no storage points then we have either a configuration error or a timing error where the library flywheel-common has not registered entry points before we init storage.  In this this case I would rather loop over an array of 0, very minimal cost, and load the storage points on the next instantiation.   

But really this case should never exist so it might be better to just throw an error and abort. If we are initing storage we should have at least one entrypoint registered. Thoughts?

---

I think it's fine as is